superset-frontend/src/cccs-viz/plugins/plugin-chart-ag-grid/src/ag-grid/AGGridViz.tsx

```diff
@@ -466,14 +466,22 @@ export default function AGGridViz({
       selectedData.typeData.eml_path &&
       selectedData.typeData.eml_path.length > 0
     ) {
+      const uniqueEmlPaths = new Set(selectedData.typeData.eml_path);
       specialMenuItems = [
         ...specialMenuItems,
         <DownloadEmailMenuItem
           onSelection={handleContextMenu}
-          label="Download Email"
-          disabled={Object.keys(selectedData.highlightedData).length !== 1}
+          label={
+            uniqueEmlPaths.size > 1 ? 'Download EML files' : 'Download EML file'
+          }
           key="download-email"
-          data={selectedData.typeData.eml_path[0]}
+          data={Array.from(uniqueEmlPaths)}
+          disabled={uniqueEmlPaths.size > SUBMISSION_LIMIT}
+          tooltip={
```

Comment:
Comment for the tooltip and the condition at line 479. Do we want to limit of email downloads? If so, we shouldn't reuse the same limit as AssemblyLine submission, it should be a separate constant because chances are it will be a different value.

Reply:
What would the value be may I ask?  I feel we should limit the downloads.

Reply:
Right, we should probably set a limit... @cccs-rc, any use case or requirements?

Reply:
I don't have any hard requirements. Let's say 10 and wait for requests to change?

Reply:
I'm easy, 10 is good for now, just use a different constant with a name specific to this purpose.